File: ios/MonProfPerso/Screens/AdminScreens.swift
import SwiftUI

/* ====================================================================== *
 * ÉCRAN ADMIN — GÉRER LE CATALOGUE (matières & niveaux)
 * Réservé au rôle admin ; les écritures appellent les routes /api/admin/… en live.
 * Pendant iOS du AdminCatalogScreen Android (mêmes endpoints, même UI).
 * ====================================================================== */

/// Translittère un libellé en slug ASCII (« Espagnol » → « espagnol »).
private func slugify(_ input: String) -> String {
    let folded = input.folding(options: .diacriticInsensitive, locale: .current).lowercased()
    let mapped = folded.map { ch -> Character in
        (ch.isLetter && ch.isASCII) || ch.isNumber ? ch : "-"
    }
    let collapsed = String(mapped).split(separator: "-", omittingEmptySubsequences: true).joined(separator: "-")
    return collapsed
}

struct AdminCatalogScreen: View {
    @EnvironmentObject var router: Router
    @State private var subjects: [SubjectDTO] = []
    @State private var levels: [LevelDTO] = []
    @State private var newSubject = ""
    @State private var newSubjectAccent = "green"
    @State private var newLevel = ""
    @State private var message: String? = nil

    var body: some View {
        AkScreen {
            TopBar(title: "Gérer le catalogue", subtitle: "Espace administrateur", onBack: { router.back() })
            ScrollView {
                VStack(alignment: .leading, spacing: 0) {

                    // ---------------------------------------------------- Matières
                    sectionTitle("Matières", count: subjects.count)
                    adminField(text: $newSubject, placeholder: "Ex. Musique, Espagnol…").padding(.top, 10)
                    HStack(spacing: 8) {
                        Text("Couleur").font(AkFont.regular(12.5)).foregroundColor(Ak.muted)
                        accentDot("green")
                        accentDot("orange")
                        Spacer()
                        addButton(enabled: !newSubject.trimmed.isEmpty, action: addSubject)
                    }.padding(.top, 8)

                    VStack(spacing: 9) {
                        ForEach(subjects) { s in
                            catalogRow(name: s.name, slug: s.slug, accent: s.accent) { deleteSubject(s) }
                        }
                    }.padding(.top, 14)

                    // ---------------------------------------------------- Niveaux
                    sectionTitle("Niveaux", count: levels.count).padding(.top, 22)
                    HStack(spacing: 10) {
                        adminField(text: $newLevel, placeholder: "Ex. Université, Master…")
                        addButton(enabled: !newLevel.trimmed.isEmpty, action: addLevel)
                    }.padding(.top, 10)

                    VStack(spacing: 9) {
                        ForEach(levels) { l in
                            catalogRow(name: l.name, slug: l.slug, accent: "green") { deleteLevel(l) }
                        }
                    }.padding(.top, 14)

                    if let message {
                        Text(message).font(AkFont.semibold(12.5)).foregroundColor(Ak.green).padding(.top, 16)
                    }
                    Spacer(minLength: 20)
                }
                .padding(.horizontal, 22).padding(.top, 8)
            }
        }
        .task { await reload() }
    }

    // MARK: Données
    private func reload() async {
        if let s = try? await ApiClient.shared.subjects() { subjects = s }
        if let l = try? await ApiClient.shared.levels() { levels = l }
    }

    private func addSubject() {
        let slug = slugify(newSubject)
        guard !slug.isEmpty else { return }
        Task { @MainActor in
            do {
                _ = try await ApiClient.shared.createSubject(slug: slug, name: newSubject.trimmed, accent: newSubjectAccent)
                newSubject = ""; message = "Matière ajoutée"; await reload()
            } catch { message = "Échec : matière déjà existante ?" }
        }
    }

    private func deleteSubject(_ s: SubjectDTO) {
        Task { @MainActor in
            if (try? await ApiClient.shared.deleteSubject(slug: s.slug)) != nil { message = "Matière supprimée"; await reload() }
        }
    }

    private func addLevel() {
        let slug = slugify(newLevel)
        guard !slug.isEmpty else { return }
        Task { @MainActor in
            do {
                _ = try await ApiClient.shared.createLevel(slug: slug, name: newLevel.trimmed, ord: levels.count + 1)
                newLevel = ""; message = "Niveau ajouté"; await reload()
            } catch { message = "Échec : niveau déjà existant ?" }
        }
    }

    private func deleteLevel(_ l: LevelDTO) {
        Task { @MainActor in
            if (try? await ApiClient.shared.deleteLevel(slug: l.slug)) != nil { message = "Niveau supprimé"; await reload() }
        }
    }

    // MARK: Sous-vues
    private func sectionTitle(_ title: String, count: Int) -> some View {
        HStack(spacing: 8) {
            Text(title).font(AkFont.schibstedExtra(18)).foregroundColor(Ak.ink)
            Text("\(count)").font(AkFont.bold(11.5)).foregroundColor(Ak.green)
                .padding(.horizontal, 8).padding(.vertical, 3)
                .background(Ak.greenSoft).clipShape(RoundedRectangle(cornerRadius: 8))
        }
    }

    private func adminField(text: Binding<String>, placeholder: String) -> some View {
        TextField("", text: text, prompt: Text(placeholder).foregroundColor(Ak.faint))
            .font(AkFont.regular(14)).foregroundColor(Ak.ink).tint(Ak.green)
            .padding(.horizontal, 14).padding(.vertical, 13)
            .background(.white).clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
            .overlay(RoundedRectangle(cornerRadius: 14).stroke(Ak.border, lineWidth: 1))
    }

    private func accentDot(_ accent: String) -> some View {
        let selected = newSubjectAccent == accent
        let c = accent == "orange" ? Ak.orange : Ak.green
        return RoundedRectangle(cornerRadius: 8, style: .continuous).fill(c).frame(width: 26, height: 26)
            .overlay(RoundedRectangle(cornerRadius: 8).stroke(Ak.ink, lineWidth: selected ? 2 : 0))
            .contentShape(Rectangle()).onTapGesture { newSubjectAccent = accent }
    }

    private func addButton(enabled: Bool, action: @escaping () -> Void) -> some View {
        HStack(spacing: 4) {
            Image(systemName: "plus").font(.system(size: 13, weight: .bold))
            Text("Ajouter").font(AkFont.bold(13))
        }
        .foregroundColor(.white)
        .padding(.horizontal, 16).padding(.vertical, 11)
        .background(enabled ? Ak.green : Ak.border)
        .clipShape(RoundedRectangle(cornerRadius: 12, style: .continuous))
        .contentShape(Rectangle())
        .onTapGesture { if enabled { action() } }
    }

    private func catalogRow(name: String, slug: String, accent: String, onDelete: @escaping () -> Void) -> some View {
        let dot = accent == "orange" ? Ak.orange : Ak.green
        return HStack(spacing: 12) {
            RoundedRectangle(cornerRadius: 3, style: .continuous).fill(dot).frame(width: 10, height: 10)
            VStack(alignment: .leading, spacing: 1) {
                Text(name).font(AkFont.bold(14)).foregroundColor(Ak.ink)
                Text(slug).font(AkFont.regular(11.5)).foregroundColor(Ak.faint)
            }
            Spacer()
            Image(systemName: "xmark").font(.system(size: 13, weight: .bold)).foregroundColor(Ak.orange)
                .frame(width: 30, height: 30).background(Ak.orangeSoft)
                .clipShape(RoundedRectangle(cornerRadius: 9, style: .continuous))
                .contentShape(Rectangle()).onTapGesture(perform: onDelete)
        }
        .padding(.horizontal, 14).padding(.vertical, 12)
        .background(.white).clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
        .overlay(RoundedRectangle(cornerRadius: 14).stroke(Ak.border, lineWidth: 1))
    }
}

/* ====================================================================== *
 * ÉCRAN ADMIN — RESSOURCES PÉDAGOGIQUES (cours / devoirs / exercices)
 * Création de métadonnées (titre, matière, niveau, description) ; le fichier
 * reste optionnel côté API. Écritures en live sur les routes /api/admin/resources.
 * ====================================================================== */
private let resourceTypes: [(value: String, label: String)] = [
    ("course", "Cours"), ("homework", "Devoir"), ("exercise", "Exercice"),
]
private func resourceTypeLabel(_ type: String) -> String {
    resourceTypes.first { $0.value == type }?.label ?? type
}

struct AdminResourcesScreen: View {
    @EnvironmentObject var router: Router
    @State private var subjects: [SubjectDTO] = []
    @State private var levels: [LevelDTO] = []
    @State private var resources: [ResourceDTO] = []
    @State private var type = "course"
    @State private var title = ""
    @State private var subjectSlug: String? = nil
    @State private var level: String? = nil
    @State private var desc = ""
    @State private var message: String? = nil

    var body: some View {
        AkScreen {
            TopBar(title: "Ressources pédagogiques", subtitle: "Espace administrateur", onBack: { router.back() })
            ScrollView {
                VStack(alignment: .leading, spacing: 0) {

                    fieldLabel("Type")
                    ScrollView(.horizontal, showsIndicators: false) {
                        HStack(spacing: 8) {
                            ForEach(resourceTypes, id: \.value) { t in
                                pickChip(t.label, selected: type == t.value) { type = t.value }
                            }
                        }
                    }.padding(.top, 8)

                    fieldLabel("Titre").padding(.top, 14)
                    adminField(text: $title, placeholder: "Ex. Fiche de révision — Théorème de Thalès").padding(.top, 8)

                    fieldLabel("Matière (facultatif)").padding(.top, 14)
                    ScrollView(.horizontal, showsIndicators: false) {
                        HStack(spacing: 8) {
                            pickChip("Aucune", selected: subjectSlug == nil) { subjectSlug = nil }
                            ForEach(subjects) { s in
                                pickChip(s.name, selected: subjectSlug == s.slug) { subjectSlug = s.slug }
                            }
                        }
                    }.padding(.top, 8)

                    fieldLabel("Niveau (facultatif)").padding(.top, 14)
                    ScrollView(.horizontal, showsIndicators: false) {
                        HStack(spacing: 8) {
                            pickChip("Aucun", selected: level == nil) { level = nil }
                            ForEach(levels) { l in
                                pickChip(l.name, selected: level == l.slug) { level = l.slug }
                            }
                        }
                    }.padding(.top, 8)

                    fieldLabel("Description (facultatif)").padding(.top, 14)
                    descField(text: $desc, placeholder: "Quelques mots sur la ressource…").padding(.top, 8)

                    HStack {
                        Spacer()
                        addButton(enabled: !title.trimmed.isEmpty, action: addResource)
                    }.padding(.top, 14)

                    sectionTitle("Ressources", count: resources.count).padding(.top, 22)
                    if resources.isEmpty {
                        Text("Aucune ressource pour l'instant.").font(AkFont.regular(13)).foregroundColor(Ak.faint).padding(.top, 12)
                    }
                    VStack(spacing: 9) {
                        ForEach(resources) { r in resourceRow(r) }
                    }.padding(.top, 12)

                    if let message {
                        Text(message).font(AkFont.semibold(12.5)).foregroundColor(Ak.green).padding(.top, 16)
                    }
                    Spacer(minLength: 20)
                }
                .padding(.horizontal, 22).padding(.top, 8)
            }
        }
        .task { await reload() }
    }

    private func reload() async {
        if let s = try? await ApiClient.shared.subjects() { subjects = s }
        if let l = try? await ApiClient.shared.levels() { levels = l }
        if let r = try? await ApiClient.shared.resources() { resources = r }
    }

    private func addResource() {
        Task { @MainActor in
            do {
                _ = try await ApiClient.shared.createResource(type: type, title: title.trimmed,
                                                              subjectSlug: subjectSlug, level: level, description: desc.trimmed)
                title = ""; desc = ""; subjectSlug = nil; level = nil
                message = "Ressource ajoutée"; await reload()
            } catch { message = "Échec de l'ajout" }
        }
    }

    private func deleteResource(_ r: ResourceDTO) {
        Task { @MainActor in
            if (try? await ApiClient.shared.deleteResource(id: r.id)) != nil { message = "Ressource supprimée"; await reload() }
        }
    }

    // MARK: Sous-vues
    private func fieldLabel(_ text: String) -> some View {
        Text(text).font(AkFont.regular(12.5)).foregroundColor(Ak.muted)
    }

    private func adminField(text: Binding<String>, placeholder: String) -> some View {
        TextField("", text: text, prompt: Text(placeholder).foregroundColor(Ak.faint))
            .font(AkFont.regular(14)).foregroundColor(Ak.ink).tint(Ak.green)
            .padding(.horizontal, 14).padding(.vertical, 13)
            .background(.white).clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
            .overlay(RoundedRectangle(cornerRadius: 14).stroke(Ak.border, lineWidth: 1))
    }

    private func descField(text: Binding<String>, placeholder: String) -> some View {
        TextField("", text: text, prompt: Text(placeholder).foregroundColor(Ak.faint), axis: .vertical)
            .lineLimit(3...5)
            .font(AkFont.regular(14)).foregroundColor(Ak.ink).tint(Ak.green)
            .padding(.horizontal, 14).padding(.vertical, 13)
            .background(.white).clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
            .overlay(RoundedRectangle(cornerRadius: 14).stroke(Ak.border, lineWidth: 1))
    }

    private func pickChip(_ label: String, selected: Bool, action: @escaping () -> Void) -> some View {
        Text(label).font(AkFont.bold(12.5)).foregroundColor(selected ? .white : Ak.inkSoft)
            .padding(.horizontal, 14).padding(.vertical, 9)
            .background(selected ? Ak.green : .white)
            .clipShape(RoundedRectangle(cornerRadius: 11, style: .continuous))
            .overlay(RoundedRectangle(cornerRadius: 11).stroke(Ak.border, lineWidth: selected ? 0 : 1))
            .contentShape(Rectangle()).onTapGesture(perform: action)
    }

    private func sectionTitle(_ title: String, count: Int) -> some View {
        HStack(spacing: 8) {
            Text(title).font(AkFont.schibstedExtra(18)).foregroundColor(Ak.ink)
            Text("\(count)").font(AkFont.bold(11.5)).foregroundColor(Ak.green)
                .padding(.horizontal, 8).padding(.vertical, 3)
                .background(Ak.greenSoft).clipShape(RoundedRectangle(cornerRadius: 8))
        }
    }

    private func addButton(enabled: Bool, action: @escaping () -> Void) -> some View {
        HStack(spacing: 4) {
            Image(systemName: "plus").font(.system(size: 13, weight: .bold))
            Text("Ajouter").font(AkFont.bold(13))
        }
        .foregroundColor(.white)
        .padding(.horizontal, 16).padding(.vertical, 11)
        .background(enabled ? Ak.green : Ak.border)
        .clipShape(RoundedRectangle(cornerRadius: 12, style: .continuous))
        .contentShape(Rectangle())
        .onTapGesture { if enabled { action() } }
    }

    private func resourceRow(_ r: ResourceDTO) -> some View {
        let accent = r.type == "homework" ? Ak.orange : Ak.green
        let soft = r.type == "homework" ? Ak.orangeSoft : Ak.greenSoft
        let tags = [r.subject_slug, r.level].compactMap { $0 }.joined(separator: " · ")
        return HStack(spacing: 10) {
            VStack(alignment: .leading, spacing: 4) {
                HStack(spacing: 8) {
                    Text(resourceTypeLabel(r.type)).font(AkFont.bold(10.5)).foregroundColor(accent)
                        .padding(.horizontal, 7).padding(.vertical, 2)
                        .background(soft).clipShape(RoundedRectangle(cornerRadius: 7))
                    if !tags.isEmpty { Text(tags).font(AkFont.regular(11)).foregroundColor(Ak.faint) }
                }
                Text(r.title).font(AkFont.bold(14)).foregroundColor(Ak.ink)
                if let d = r.description, !d.isEmpty {
                    Text(d).font(AkFont.regular(11.5)).foregroundColor(Ak.muted)
                }
            }
            Spacer()
            Image(systemName: "xmark").font(.system(size: 13, weight: .bold)).foregroundColor(Ak.orange)
                .frame(width: 30, height: 30).background(Ak.orangeSoft)
                .clipShape(RoundedRectangle(cornerRadius: 9, style: .continuous))
                .contentShape(Rectangle()).onTapGesture { deleteResource(r) }
        }
        .padding(.horizontal, 14).padding(.vertical, 12)
        .background(.white).clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
        .overlay(RoundedRectangle(cornerRadius: 14).stroke(Ak.border, lineWidth: 1))
    }
}

private extension String {
    var trimmed: String { trimmingCharacters(in: .whitespacesAndNewlines) }
}

File: android/app/src/main/java/ci/monprofperso/app/ui/screens/AdminScreens.kt
package ci.monprofperso.app.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.Close
import androidx.compose.material3.Icon
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.OutlinedTextFieldDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.ImeAction
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import ci.monprofperso.app.data.Api
import ci.monprofperso.app.data.LevelDto
import ci.monprofperso.app.data.ResourceDto
import ci.monprofperso.app.data.SubjectDto
import ci.monprofperso.app.nav.NavActions
import ci.monprofperso.app.ui.components.*
import ci.monprofperso.app.ui.theme.AkColors
import ci.monprofperso.app.ui.theme.Hanken
import ci.monprofperso.app.ui.theme.Schibsted
import kotlinx.coroutines.launch

/* ====================================================================== *
 * ÉCRAN ADMIN — GÉRER LE CATALOGUE (matières & niveaux)
 * Réservé au rôle admin ; les écritures appellent les routes /api/admin/… en live.
 * ====================================================================== */
private fun slugify(input: String): String {
    val n = java.text.Normalizer.normalize(input, java.text.Normalizer.Form.NFD).replace(Regex("\\p{Mn}+"), "")
    return n.lowercase().replace(Regex("[^a-z0-9]+"), "-").trim('-')
}

@Composable
fun AdminCatalogScreen(nav: NavActions) {
    val scope = rememberCoroutineScope()
    var subjects by remember { mutableStateOf(listOf<SubjectDto>()) }
    var levels by remember { mutableStateOf(listOf<LevelDto>()) }
    var newSubject by remember { mutableStateOf("") }
    var newSubjectAccent by remember { mutableStateOf("green") }
    var newLevel by remember { mutableStateOf("") }
    var message by remember { mutableStateOf<String?>(null) }

    suspend fun reload() {
        runCatching { Api.service.subjects() }.onSuccess { subjects = it }
        runCatching { Api.service.levels() }.onSuccess { levels = it }
    }
    LaunchedEffect(Unit) { reload() }

    AkScreen(applyBottomInset = false) {
        TopBar("Gérer le catalogue", subtitle = "Espace administrateur", onBack = { nav.back() })
        Column(Modifier.weight(1f).verticalScrollSafe().padding(horizontal = 22.dp).padding(top = 8.dp)) {

            // ---------------------------------------------------------- Matières
            SectionTitle("Matières", "${subjects.size}")
            Spacer(Modifier.height(10.dp))
            AdminField(value = newSubject, onValueChange = { newSubject = it }, placeholder = "Ex. Musique, Espagnol…")
            Spacer(Modifier.height(8.dp))
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text("Couleur", fontFamily = Hanken, fontSize = 12.5.sp, color = AkColors.Muted)
                Spacer(Modifier.width(10.dp))
                AccentDot("green", newSubjectAccent == "green") { newSubjectAccent = "green" }
                Spacer(Modifier.width(8.dp))
                AccentDot("orange", newSubjectAccent == "orange") { newSubjectAccent = "orange" }
                Spacer(Modifier.weight(1f))
                AddButton(enabled = newSubject.isNotBlank()) {
                    val slug = slugify(newSubject)
                    if (slug.isBlank()) return@AddButton
                    scope.launch {
                        runCatching {
                            Api.service.createSubject(mapOf("slug" to slug, "name" to newSubject.trim(),
                                "accent" to newSubjectAccent, "icon" to "more"))
                        }.onSuccess { newSubject = ""; message = "Matière ajoutée"; reload() }
                            .onFailure { message = "Échec : matière déjà existante ?" }
                    }
                }
            }
            Spacer(Modifier.height(14.dp))
            subjects.forEach { s ->
                CatalogRow(s.name, s.slug, s.accent) {
                    scope.launch { runCatching { Api.service.deleteSubject(s.slug) }.onSuccess { message = "Matière supprimée"; reload() } }
                }
            }

            Spacer(Modifier.height(22.dp))

            // ---------------------------------------------------------- Niveaux
            SectionTitle("Niveaux", "${levels.size}")
            Spacer(Modifier.height(10.dp))
            Row(verticalAlignment = Alignment.CenterVertically) {
                AdminField(value = newLevel, onValueChange = { newLevel = it }, placeholder = "Ex. Université, Master…",
                    modifier = Modifier.weight(1f))
                Spacer(Modifier.width(10.dp))
                AddButton(enabled = newLevel.isNotBlank()) {
                    val slug = slugify(newLevel)
                    if (slug.isBlank()) return@AddButton
                    scope.launch {
                        runCatching {
                            Api.service.createLevel(mapOf("slug" to slug, "name" to newLevel.trim(), "ord" to levels.size + 1))
                        }.onSuccess { newLevel = ""; message = "Niveau ajouté"; reload() }
                            .onFailure { message = "Échec : niveau déjà existant ?" }
                    }
                }
            }
            Spacer(Modifier.height(14.dp))
            levels.forEach { l ->
                CatalogRow(l.name, l.slug, "green") {
                    scope.launch { runCatching { Api.service.deleteLevel(l.slug) }.onSuccess { message = "Niveau supprimé"; reload() } }
                }
            }

            message?.let {
                Spacer(Modifier.height(16.dp))
                Text(it, fontFamily = Hanken, fontWeight = FontWeight.SemiBold, fontSize = 12.5.sp, color = AkColors.Green)
            }
            Spacer(Modifier.height(20.dp))
        }
    }
}

@Composable
private fun SectionTitle(title: String, count: String) {
    Row(verticalAlignment = Alignment.CenterVertically) {
        Text(title, fontFamily = Schibsted, fontWeight = FontWeight.ExtraBold, fontSize = 18.sp, color = AkColors.Ink)
        Spacer(Modifier.width(8.dp))
        Text(count, fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 11.5.sp, color = AkColors.Green,
            modifier = Modifier.clip(RoundedCornerShape(8.dp)).background(AkColors.GreenSoft).padding(horizontal = 8.dp, vertical = 3.dp))
    }
}

@Composable
private fun AdminField(value: String, onValueChange: (String) -> Unit, placeholder: String, modifier: Modifier = Modifier, singleLine: Boolean = true) {
    OutlinedTextField(
        value = value, onValueChange = onValueChange, modifier = modifier.fillMaxWidth(),
        placeholder = { Text(placeholder, fontFamily = Hanken, fontSize = 14.sp, color = AkColors.Faint) },
        singleLine = singleLine,
        minLines = if (singleLine) 1 else 3,
        keyboardOptions = KeyboardOptions(imeAction = ImeAction.Done),
        shape = RoundedCornerShape(14.dp),
        colors = OutlinedTextFieldDefaults.colors(
            focusedBorderColor = AkColors.Green, unfocusedBorderColor = AkColors.Border,
            focusedContainerColor = AkColors.White, unfocusedContainerColor = AkColors.White,
            cursorColor = AkColors.Green,
        ),
    )
}

@Composable
private fun AccentDot(accent: String, selected: Boolean, onClick: () -> Unit) {
    val c = if (accent == "orange") AkColors.Orange else AkColors.Green
    Box(
        Modifier.size(26.dp).clip(RoundedCornerShape(8.dp)).background(c)
            .then(if (selected) Modifier.border(2.dp, AkColors.Ink, RoundedCornerShape(8.dp)) else Modifier)
            .clickable { onClick() }
    )
}

@Composable
private fun AddButton(enabled: Boolean, onClick: () -> Unit) {
    val bg = if (enabled) AkColors.Green else AkColors.Border
    Row(
        Modifier.clip(RoundedCornerShape(12.dp)).background(bg).clickable(enabled = enabled) { onClick() }
            .padding(horizontal = 16.dp, vertical = 11.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Icon(Icons.Filled.Add, null, tint = AkColors.White, modifier = Modifier.size(17.dp))
        Text(" Ajouter", fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 13.sp, color = AkColors.White)
    }
}

/* ====================================================================== *
 * ÉCRAN ADMIN — RESSOURCES PÉDAGOGIQUES (cours / devoirs / exercices)
 * Création de métadonnées (titre, matière, niveau, description) ; le fichier
 * reste optionnel côté API. Écritures en live sur /api/admin/resources.
 * ====================================================================== */
private val RESOURCE_TYPES = listOf("course" to "Cours", "homework" to "Devoir", "exercise" to "Exercice")

private fun typeLabel(type: String): String = RESOURCE_TYPES.firstOrNull { it.first == type }?.second ?: type

@Composable
fun AdminResourcesScreen(nav: NavActions) {
    val scope = rememberCoroutineScope()
    var subjects by remember { mutableStateOf(listOf<SubjectDto>()) }
    var levels by remember { mutableStateOf(listOf<LevelDto>()) }
    var resources by remember { mutableStateOf(listOf<ResourceDto>()) }
    var type by remember { mutableStateOf("course") }
    var title by remember { mutableStateOf("") }
    var subjectSlug by remember { mutableStateOf<String?>(null) }
    var level by remember { mutableStateOf<String?>(null) }
    var description by remember { mutableStateOf("") }
    var message by remember { mutableStateOf<String?>(null) }

    suspend fun reload() {
        runCatching { Api.service.subjects() }.onSuccess { subjects = it }
        runCatching { Api.service.levels() }.onSuccess { levels = it }
        runCatching { Api.service.resources() }.onSuccess { resources = it }
    }
    LaunchedEffect(Unit) { reload() }

    AkScreen(applyBottomInset = false) {
        TopBar("Ressources pédagogiques", subtitle = "Espace administrateur", onBack = { nav.back() })
        Column(Modifier.weight(1f).verticalScrollSafe().padding(horizontal = 22.dp).padding(top = 8.dp)) {

            Text("Type", fontFamily = Hanken, fontSize = 12.5.sp, color = AkColors.Muted)
            Spacer(Modifier.height(8.dp))
            Row(Modifier.horizontalScroll(rememberScrollState())) {
                RESOURCE_TYPES.forEach { (value, label) ->
                    PickChip(label, value == type) { type = value }
                    Spacer(Modifier.width(8.dp))
                }
            }
            Spacer(Modifier.height(14.dp))

            Text("Titre", fontFamily = Hanken, fontSize = 12.5.sp, color = AkColors.Muted)
            Spacer(Modifier.height(8.dp))
            AdminField(value = title, onValueChange = { title = it }, placeholder = "Ex. Fiche de révision — Théorème de Thalès")
            Spacer(Modifier.height(14.dp))

            Text("Matière (facultatif)", fontFamily = Hanken, fontSize = 12.5.sp, color = AkColors.Muted)
            Spacer(Modifier.height(8.dp))
            Row(Modifier.horizontalScroll(rememberScrollState())) {
                PickChip("Aucune", subjectSlug == null) { subjectSlug = null }
                Spacer(Modifier.width(8.dp))
                subjects.forEach { s ->
                    PickChip(s.name, subjectSlug == s.slug) { subjectSlug = s.slug }
                    Spacer(Modifier.width(8.dp))
                }
            }
            Spacer(Modifier.height(14.dp))

            Text("Niveau (facultatif)", fontFamily = Hanken, fontSize = 12.5.sp, color = AkColors.Muted)
            Spacer(Modifier.height(8.dp))
            Row(Modifier.horizontalScroll(rememberScrollState())) {
                PickChip("Aucun", level == null) { level = null }
                Spacer(Modifier.width(8.dp))
                levels.forEach { l ->
                    PickChip(l.name, level == l.slug) { level = l.slug }
                    Spacer(Modifier.width(8.dp))
                }
            }
            Spacer(Modifier.height(14.dp))

            Text("Description (facultatif)", fontFamily = Hanken, fontSize = 12.5.sp, color = AkColors.Muted)
            Spacer(Modifier.height(8.dp))
            AdminField(value = description, onValueChange = { description = it }, placeholder = "Quelques mots sur la ressource…",
                singleLine = false)
            Spacer(Modifier.height(14.dp))

            AddButton(enabled = title.isNotBlank()) {
                scope.launch {
                    runCatching {
                        Api.service.createResource(buildMap {
                            put("type", type); put("title", title.trim())
                            subjectSlug?.let { put("subjectSlug", it) }
                            level?.let { put("level", it) }
                            if (description.isNotBlank()) put("description", description.trim())
                        })
                    }.onSuccess {
                        title = ""; description = ""; subjectSlug = null; level = null
                        message = "Ressource ajoutée"; reload()
                    }.onFailure { message = "Échec de l'ajout" }
                }
            }

            Spacer(Modifier.height(22.dp))
            SectionTitle("Ressources", "${resources.size}")
            Spacer(Modifier.height(12.dp))
            if (resources.isEmpty()) {
                Text("Aucune ressource pour l'instant.", fontFamily = Hanken, fontSize = 13.sp, color = AkColors.Faint)
            }
            resources.forEach { r -> ResourceRow(r) {
                scope.launch { runCatching { Api.service.deleteResource(r.id) }.onSuccess { message = "Ressource supprimée"; reload() } }
            } }

            message?.let {
                Spacer(Modifier.height(16.dp))
                Text(it, fontFamily = Hanken, fontWeight = FontWeight.SemiBold, fontSize = 12.5.sp, color = AkColors.Green)
            }
            Spacer(Modifier.height(20.dp))
        }
    }
}

@Composable
private fun PickChip(label: String, selected: Boolean, onClick: () -> Unit) {
    val bg = if (selected) AkColors.Green else AkColors.White
    val fg = if (selected) AkColors.White else AkColors.InkSoft
    Text(
        label, fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 12.5.sp, color = fg,
        modifier = Modifier.clip(RoundedCornerShape(11.dp)).background(bg)
            .then(if (selected) Modifier else Modifier.border(1.dp, AkColors.Border, RoundedCornerShape(11.dp)))
            .clickable { onClick() }.padding(horizontal = 14.dp, vertical = 9.dp),
    )
}

@Composable
private fun ResourceRow(r: ResourceDto, onDelete: () -> Unit) {
    val accent = if (r.type == "homework") AkColors.Orange else AkColors.Green
    val soft = if (r.type == "homework") AkColors.OrangeSoft else AkColors.GreenSoft
    Row(
        Modifier.fillMaxWidth().padding(bottom = 9.dp).clip(RoundedCornerShape(14.dp)).background(AkColors.White)
            .border(1.dp, AkColors.Border, RoundedCornerShape(14.dp)).padding(horizontal = 14.dp, vertical = 12.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Column(Modifier.weight(1f)) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text(typeLabel(r.type), fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 10.5.sp, color = accent,
                    modifier = Modifier.clip(RoundedCornerShape(7.dp)).background(soft).padding(horizontal = 7.dp, vertical = 2.dp))
                val tags = listOfNotNull(r.subjectSlug, r.level).joinToString(" · ")
                if (tags.isNotEmpty()) {
                    Spacer(Modifier.width(8.dp))
                    Text(tags, fontFamily = Hanken, fontSize = 11.sp, color = AkColors.Faint)
                }
            }
            Spacer(Modifier.height(4.dp))
            Text(r.title, fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 14.sp, color = AkColors.Ink)
            r.description?.takeIf { it.isNotBlank() }?.let {
                Text(it, fontFamily = Hanken, fontSize = 11.5.sp, color = AkColors.Muted, modifier = Modifier.padding(top = 2.dp))
            }
        }
        Spacer(Modifier.width(10.dp))
        Box(
            Modifier.size(30.dp).clip(RoundedCornerShape(9.dp)).background(AkColors.OrangeSoft).clickable { onDelete() },
            contentAlignment = Alignment.Center,
        ) { Icon(Icons.Filled.Close, "Supprimer", tint = AkColors.Orange, modifier = Modifier.size(16.dp)) }
    }
}

@Composable
private fun CatalogRow(name: String, slug: String, accent: String, onDelete: () -> Unit) {
    val dot = if (accent == "orange") AkColors.Orange else AkColors.Green
    Row(
        Modifier.fillMaxWidth().padding(bottom = 9.dp).clip(RoundedCornerShape(14.dp)).background(AkColors.White)
            .border(1.dp, AkColors.Border, RoundedCornerShape(14.dp)).padding(horizontal = 14.dp, vertical = 12.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Box(Modifier.size(10.dp).clip(RoundedCornerShape(3.dp)).background(dot))
        Spacer(Modifier.width(12.dp))
        Column(Modifier.weight(1f)) {
            Text(name, fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 14.sp, color = AkColors.Ink)
            Text(slug, fontFamily = Hanken, fontSize = 11.5.sp, color = AkColors.Faint)
        }
        Box(
            Modifier.size(30.dp).clip(RoundedCornerShape(9.dp)).background(AkColors.OrangeSoft).clickable { onDelete() },
            contentAlignment = Alignment.Center,
        ) { Icon(Icons.Filled.Close, "Supprimer", tint = AkColors.Orange, modifier = Modifier.size(16.dp)) }
    }
}
